packages/java/endpoint/src/main/java/com/vaadin/hilla/route/ClientRouteRegistry.java

```diff
@@ -84,11 +87,34 @@ public void removeRoute(String route) {
     public ClientViewConfig getRouteByPath(String path) {
         final Set<String> routes = registeredRoutes.keySet();
         final AntPathMatcher pathMatcher = new AntPathMatcher();
-        for (String route : routes) {
-            if (pathMatcher.match(route, path)) {
-                return registeredRoutes.get(route);
-            }
+        return Stream.of(addTrailingSlash(path), removeTrailingSlash(path))
+                .map(p -> {
+                    for (String route : routes) {
+                        if (pathMatcher.match(route, p)) {
+                            return registeredRoutes.get(route);
+                        }
+                    }
+                    return null;
+                }).filter(Objects::nonNull).findFirst().orElse(null);
+    }
+
+    private String addTrailingSlash(String path) {
+        if (path.isEmpty()) {
+            return "/";
+        }
+        if (path.charAt(path.length() - 1) != '/') {
+            return path + "/";
+        }
+        return path;
+    }
+
+    private String removeTrailingSlash(String path) {
+        if (path.isEmpty()) {
+            return "";
+        }
+        if (path.charAt(path.length() - 1) == '/') {
```

Comment:
Any specific reason for not using
```suggestion
        if (path.endsWith("/")) {
```